modin/pandas/dataframe.py

```diff
@@ -369,12 +370,15 @@ def groupby(
         elif hashable(by) and not isinstance(by, pandas.Grouper):
             drop = by in self.columns
             idx_name = by
-            if self._query_compiler.has_multiindex(
-                axis=axis
-            ) and by in self._query_compiler.get_index_names(axis):
+            if (
+                self._query_compiler.has_multiindex(axis=axis)
+                and by in self._query_compiler.get_index_names(axis)
+                and by is not None
+            ):
                 # In this case we pass the string value of the name through to the
                 # partitions. This is more efficient than broadcasting the values.
-                pass
+                level = by
+                by = None
```

Comment:
if `by` is a scalar and it's an index level name, moving `by` content to `level` and setting `by` as None

Reply:
It's still not clear to me why we need to manipulate `by` and `level` if we are passing both to the `query_compiler` and eventually the worker partitions.

Reply:
there could be cases (described [here](https://github.com/modin-project/modin/pull/2790#issue-581022973)) when we need to distinguish column-by and level-by, as I mentioned before it's not critical and we can do this splitting only in places where we need it, but to ease this process I decided to do it once at the API level.